stripe.php

```diff
@@ -42,7 +42,7 @@ function stripe_civicrm_install() {
     `plan_id` varchar(255) COLLATE utf8_unicode_ci NOT NULL,
     `is_live` tinyint(4) NOT NULL COMMENT 'Whether this is a live or test transaction',
     `processor_id` int(10) DEFAULT NULL COMMENT 'ID from civicrm_payment_processor',
-    UNIQUE KEY `plan_id` (`plan_id`)
+    UNIQUE KEY `plan_id` (`plan_id`,`is_live`,`processor_id`)
```

Comment:
The motivation for the fix is because I doRecurPayment() in CRM/Core/Payment/Stripe.php first checks for a matching plan searching on plan_id, is_live and processor_id. If no matching record is found, it inserts a record. By making all three fields that are searched a unique key, it ensures we will never get a duplicate key error.

For example, if a record already exists with the same plan id and processor id but a different is_live setting, then the code would conclude that no matching record exists and try to insert it. If the plan_id unique key only covered the plan_id and the processor_id, it would detect this new record as a duplicate and complain.

Having said all of this...